File: micronaut-camunda-bpm-feature/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/DoNothingDelegate.java
package info.novatec.micronaut.camunda.bpm.feature.test;

import org.camunda.bpm.engine.delegate.DelegateExecution;
import org.camunda.bpm.engine.delegate.JavaDelegate;

import javax.inject.Named;
import javax.inject.Singleton;

/**
 * {@link JavaDelegate} which doesn't do anything. This implementation may be replaced in other tests.
 *
 * @author Tobias Schäfer
 */
@Singleton
@Named("doSomethingDelegate")
public class DoNothingDelegate implements JavaDelegate {
    @Override
    public void execute(DelegateExecution execution) {
        // Nothing per default
    }
}

File: micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/DoNothingDelegate.kt
package info.novatec.micronaut.camunda.bpm.feature.test

import org.camunda.bpm.engine.delegate.DelegateExecution
import org.camunda.bpm.engine.delegate.JavaDelegate
import javax.inject.Named
import javax.inject.Singleton

/**
 * [JavaDelegate] which doesn't do anything. This implementation may be replaced in other tests.
 *
 * @author Tobias Schäfer
 */
@Singleton
@Named("doSomethingDelegate")
class DoNothingDelegate : JavaDelegate {
    override fun execute(execution: DelegateExecution) {
        // Nothing per default
    }
}